test/commands/crashes/upload-mappings/upload-mappings-test.ts

```diff
@@ -0,0 +1,244 @@
+      // Arrange
+      const mappingsPath = await createMappingsFile(mappingsFileContent, "test.test");
+      // Act/Assert
+      await expect(executeUploadCommand(["-m", mappingsPath])).to.eventually.be.rejected;
+    });
+  });
+
+  describe("when AndroidProguard network requests are successful", () => {
+    beforeEach(() => {
+      expectedRequestsScope = _.flow(setupSuccessfulAndroidProguardPatchUploadResponse, setupSuccessfulAndroidProguardPostUploadResponse)(Nock(fakeHost));
+      skippedRequestsScope = setupSuccessfulAndroidProguardAbortUploadResponse(Nock(fakeHost));
+    });
+
+    it("uploads mapping.txt", async () => {
+      // Arrange
+      const mappingsPath = await createMappingsFile(mappingsFileContent);
+
+      // Act
+      const result = await executeUploadCommand(["-m", mappingsPath, "-n", "1.0", "-c", "1"]);
+
+      // Assert
+      testCommandSuccess(result, expectedRequestsScope, skippedRequestsScope);
+      const [url, uploadedZipPath] = AzureBlobUploadHelperMock.getUploadedArtifactUrlAndPath();
+      expect(url).to.eql(fakeFullUploadUrl, `mapping.txt file should be uploaded to ${fakeFullUploadUrl}`);
+      expect(uploadedZipPath).to.eql(mappingsPath, "mapping.txt file should be passed as it is");
+    });
+  });
+
+  describe("when AndroidProguard upload fails", () => {
+    before(() => {
+      AzureBlobUploadHelperMock.throwOnUpload = true;
+    });
+
+    beforeEach(() => {
+        expectedRequestsScope = _.flow(setupSuccessfulAndroidProguardAbortUploadResponse, setupSuccessfulAndroidProguardPostUploadResponse)(Nock(fakeHost));
+        skippedRequestsScope = setupSuccessfulAndroidProguardPatchUploadResponse(Nock(fakeHost));
+    });
+
+    it("aborts the symbol uploading", async () => {
+      // Arrange
+      const mappingsPath = await createMappingsFile(mappingsFileContent);
+
+      // Act
+      const result = await expect(executeUploadCommand(["-m", mappingsPath, "-n", "1.0", "-c", "1"])).to.eventually.be.rejected;
+
+      // Assert
+      testUploadFailure(result, expectedRequestsScope, skippedRequestsScope);
+    });
+
+    after(() => {
+      AzureBlobUploadHelperMock.throwOnUpload = false;
+    });
+  });
+
+  afterEach(() => {
+    Nock.cleanAll();
+  });
+
+  after(() => {
+    Nock.enableNetConnect();
+  });
+
+  async function executeUploadCommand(args: string[]): Promise<CommandResult> {
+    const uploadSymbolsCommand = new UploadMappingsCommand(getCommandArgs(args));
+    return await uploadSymbolsCommand.execute();
+  }
+
+  function testCommandSuccess(result: CommandResult, executionScope: Nock.Scope, abortScope: Nock.Scope) {
+      expect(result.succeeded).to.eql(true, "Command should be successfully completed");
+      expect(abortScope.isDone()).to.eql(false, "Upload should not be aborted");
+      executionScope.done(); // All normal API calls are executed
+  }
+
+  function testUploadFailure(result: CommandResult, executionScope: Nock.Scope, patchScope: Nock.Scope) {
```

Comment:
If this isn't being re-used, perhaps we should move it back into the test for the time being?

Reply:
Sounds good to me.